### PR TITLE
Add travis Windows build support

### PR DESCRIPTION
**Pull request by icanhazstring**
Trying to add windows build for php.
This will replace appveyor if possible

### CODE REVIEW

**Comment by lisachenko** [comment]
Hi, thx! Let's target 2.x branch for the PR

**Comment by lisachenko** [comment]
> The language 'php' is currently unsupported on the Windows Build Environment.
Let us know if you'd like to see it: https://travis-ci.community/c/environments/windows. Thanks for understanding!

**Comment by icanhazstring** [comment]
> > The language 'php' is currently unsupported on the Windows Build Environment.
> > Let us know if you'd like to see it: https://travis-ci.community/c/environments/windows. Thanks for understanding!

Yep. Tryping to find a workaround. Maybe it works for at least current php version. Lets see ;)

**Comment by lisachenko** [comment]
Please read https://travis-ci.community/t/where-to-contribute-php-support-for-windows/304/3 some info may be useful

**Comment by icanhazstring** [comment]
Almost there ;)

**Comment by lisachenko** [comment]
Congratulations! 🎆

**Comment by icanhazstring** [comment]
> Congratulations! 🎆

Thanks @lisachenko. There is a slight bug here comparing two strings of generated classes.
Maybe this should be checked in another issue/pr. This has something to do with heredoc on windows. (Might be using windows lineendings)

Can possibly be solved with:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/commit/d60e963e80f45e2050085964f67c6769f347381f
https://github.com/GeckoPackages/GeckoPHPUnit/pull/18/files

---
Edit:
Or maybe this should be fixed here.
Since this will break the build ;)

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
+        - COVERAGE_FLAGS="--coverage-text --coverage-clover=clover.xml"
+    - language: php
+      php: nightly
+      before_script:
+        - composer install
+    - name: "PHP: 7.2"
+      language: sh
+      os: windows
+      before_script:
+        - choco install php --version 7.2.15 --package-parameters='"/InstallDir:C:\php"'
```
Comment (lisachenko) [inline .travis.yml:20]: It would be nice to fetch latest available version of 7.2.x here to download it automatically.

**Comment by lisachenko** [comment]
Could you please also target 2.x branch? I want to make 2.x working with Windows build too. And perform merge of 2.x into master later.

File: .travis.yml [hunk 2]
```diff
@@ -1,18 +1,33 @@
...
+      env:
+        - COVERAGE_FLAGS="--coverage-text --coverage-clover=clover.xml"
+    - language: php
+      php: nightly
+      before_script:
+        - composer install
+    - name: "PHP: 7.2"
+      language: sh
+      os: windows
+      before_script:
```
Comment (lisachenko) [inline .travis.yml:19]: Curious if Travis supports color output )
```
-  SET ANSICON=121x90 (121x90)
```

File: .travis.yml (hunk 1 above)
Comment (icanhazstring) [reply .travis.yml:20]: Puh don't know if that is possible. I will look into it

Comment (icanhazstring) [reply .travis.yml:20]: Found this issue:

https://github.com/chocolatey/choco/issues/800

File: .travis.yml (hunk 2 above)
Comment (icanhazstring) [reply .travis.yml:19]: Lets see xD

**Comment by icanhazstring** [comment]
> Could you please also target 2.x branch? I want to make 2.x working with Windows build too. And perform merge of 2.x into master later.

@lisachenko rebased from 2.x

**Comment by icanhazstring** [comment]
Oh hello. The build works now :D

File: .travis.yml (hunk 2 above)
Comment (icanhazstring) [reply .travis.yml:19]: Sadly not

**Comment by icanhazstring** [comment]
@lisachenko I will squash commits into a single one. So we can merge this one a tackle #402 issue on windows.

**Comment by lisachenko** [comment]
All green! Perfect! 👍

**Pull request merged**